src/app/order-form-page/order-form-page.component.ts

```diff
@@ -1,26 +1,50 @@
-import {
-  Component,
-  ElementRef,
-  OnInit,
-  ViewChild,
-  ViewContainerRef,
-} from '@angular/core';
+import { Component, OnDestroy, OnInit, ViewChild } from '@angular/core';
+import { Subject, takeUntil } from 'rxjs';
+import { CartService } from '../service/cart.service';
 import { FoodServiceService } from '../service/food-service.service';
 import { FormComponent } from './components/form/form.component';
+import { IOrdersHistoryItem } from '../shared/types/types';
+import { OrderHistoryService } from '../service/order-history-service.service';
+import { Router } from '@angular/router';
 
 @Component({
   selector: 'food-order-form-page',
   templateUrl: './order-form-page.component.html',
   styleUrls: ['./order-form-page.component.scss'],
 })
-export class OrderFormPageComponent implements OnInit {
+export class OrderFormPageComponent implements OnInit, OnDestroy {
   @ViewChild('orderForm') myForm: FormComponent;
-  constructor(private foodServiceService: FoodServiceService) {}
 
-  ngOnInit(): void {}
+  orderList: IOrdersHistoryItem[];
+  destroyed$: Subject<boolean> = new Subject();
+
+  constructor(
+    private orderHistoryService: OrderHistoryService,
+    private cartService: CartService,
+    private router: Router
+  ) {}
+
+  ngOnInit(): void {
+    this.cartService
+      .getCartItems()
+      .pipe(takeUntil(this.destroyed$))
+      .subscribe((data) => {
+        this.orderList = data;
+      });
+  }
 
   submitHandler() {
-    this.foodServiceService.addOrderItemToHistory(this.myForm.orderForm.value);
+    this.orderHistoryService.addOrderItemToHistory({
+      ...this.myForm.orderForm.value,
+      date: new Date().toString(),
+      foods: this.orderList,
+    });
+    this.cartService.clearCart();
     this.myForm.orderForm.reset();
+    this.router.navigate(['/order-history']);
+  }
+  ngOnDestroy() {
+    this.destroyed$.next(true);
+    this.destroyed$.complete();
```

Comment:
and then call unsubsribe from orderList ? What is the reason of creating an on destroy subject?